website_sale_one_step_checkout/controllers/main.py

```diff
@@ -0,0 +1,240 @@
+
+                template = request.env['ir.ui.view'].render_template("website_sale_one_step_checkout.address",
+                                                                     render_values)
+
+                return {
+                    'success':True,
+                    'template':template,
+                    'mode':mode
+                }
+
+
+        country = 'country_id' in values and values['country_id'] != '' and request.env['res.country'].browse(
+            int(values['country_id']))
+        country = country and country.exists() or def_country_id
+        render_values = {
+            'partner_id': partner_id,
+            'mode': mode,
+            'checkout': values,
+            'country': country,
+            'countries': country.get_website_sale_countries(mode=mode[1]),
+            "states": country.get_website_sale_states(mode=mode[1]),
+            'error': errors,
+            'callback': kw.get('callback'),
+        }
+
+        template = request.env['ir.ui.view'].render_template("website_sale_one_step_checkout.checkout_new_address_modal", render_values)
+        return {
+            'success': True,
+            'template': template,
+            'type': mode[1],
+            'modal_title': _(kw.get('modal_title', ''))
+        }
+
+    @http.route(['/shop/checkout/proceed_payment/'], type='json', auth='public', website=True,
+                multilang=True)
+    def proceed_payment(self, **post):
+        # must have a draft sale order with lines at this point, otherwise redirect to shop
+        SaleOrder = request.env['sale.order']
+        order = request.website.sale_get_order()
+
+        redirection = self.checkout_redirection(order)
+        if redirection:
+            return redirection
+
+        # part from shop/confirm_order
+        order.onchange_partner_shipping_id()
+        order.order_line._compute_tax_id()
+        request.session['sale_last_order_id'] = order.id
+        extra_step = request.env.ref('website_sale.extra_info_option')
+
+        # TODO: HOW TO HANDLE THIS CASE?
+        if extra_step.active:
+            # TODO CONTROLLER /shop/extra_info NEEDS TO BE REWRITTEN?
+            return request.redirect("/shop/extra_info")
+
+        # part from /shop/payment
+        shipping_partner_id = False
+        if order:
+            if order.partner_shipping_id:
+                shipping_partner_id = order.partner_shipping_id.id
+            else:
+                shipping_partner_id = order.partner_invoice_id.id
+
+        # TODO: ADAPT BELOW PART FROM THE ORIGINAL CONTROLLER
+        values = {
+            'website_sale_order': order
+        }
+        values['errors'] = SaleOrder._get_errors(order)
+        values.update(SaleOrder._get_website_data(order))
+
+        # TODO TAKE CARE OF ERRORS
+        if values['errors']:
+            raise NotImplementedError('action must be defined!')
+
+    @http.route(['/page/terms_and_conditions/'], type='http', auth="public",
```

Comment:
Remove this part completely because we want to add the possibility to handle the whole terms and condition in a better way. And we want to add this possibility for the default checkout.